src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,263 @@
+import { v4 as uuid } from 'uuid';
+import { AuthorData, FormData } from 'helpers/Types';
+import './CreateCourse.css';
+import { formatCreationDate } from 'helpers/formatCreationDate';
+
+export const CreateCourse = ({
+  authorList,
+  updateCourses,
+  updateAuthors,
+}: any) => {
+  const initialValues = { title: '', description: '', duration: 0 };
+  const [formData, setFormData] = useState(initialValues);
+  const [courseAuthors, setCourseAuthors] = useState<
+    { id: string; name: string }[]
+  >([]);
+  const [authorsList, setAuthorsList] = useState(authorList);
+  const [name, setName] = useState('');
+  const [errorMessage, setErrorMessage] = useState({
+    title: '',
+    description: '',
+    duration: '',
+  });
+  const [authorError, setAuthorError] = useState('');
+  const responseFormBody: FormData = {
+    id: '',
+    title: '',
+    description: '',
+    duration: '',
+    creationDate: new Date().toDateString(),
+    authors: [],
+  };
+  const navigate = useNavigate();
+  const handleInputFields = (e: any) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const result = validation(formData);
+    setErrorMessage(result);
+    if (
+      !result.title.length &&
+      !result.description.length &&
+      !result.duration.length
+    ) {
+      responseFormBody.id = uuid();
+      responseFormBody.title = formData.title;
+      responseFormBody.description = formData.description;
+      responseFormBody.duration = String(formData.duration);
+      responseFormBody.creationDate = formatCreationDate(new Date().toString());
+      responseFormBody.authors = courseAuthors.map((a) => {
+        return a.id;
+      });
+      updateCourses(responseFormBody);
+      navigate('/courses');
+    }
+  };
+
+  const addToList = (e: any) => {
+    e.preventDefault();
+    if (name) {
+      const doesNameExist = authorsList.some(
+        (author: AuthorData) => author.name === name
+      );
+      if (doesNameExist) {
+        setAuthorError('Author with this name already exists!');
+        return;
+      }
+      const newAuthor = { id: uuid(), name };
+      setAuthorsList([...authorsList, newAuthor]);
+      updateAuthors(newAuthor);
+      setName('');
+    } else {
+      setAuthorError('');
+    }
+  };
+
+  const handleAddButton = (author: { id: string; name: string }) => {
+    const newAuthorsList = authorsList.filter((item: any) => {
+      return item.id !== author.id;
+    });
+    setAuthorsList(newAuthorsList);
+    setCourseAuthors([...courseAuthors, author]);
+  };
+
+  const handleDeleteButton = (author: { id: string; name: string }) => {
+    const newCourseList = courseAuthors.filter((item) => {
+      return item.id !== author.id;
+    });
+    setAuthorsList([...authorsList, author]);
+    setCourseAuthors(newCourseList);
+  };
+
+  const validation = (formData: {
```

Comment:
Please also validate the author name

![image](https://github.com/elenapetra/react-app/assets/145835962/f5a83d5d-285c-4a2a-9b01-d63274645352)

(from https://d17btkcdsmqrmh.cloudfront.net/react-fundamentals/docs/module-2/home-task/components#create-createcourse-component)